sriovnet_switchdev.go

```diff
@@ -226,6 +226,44 @@ func findNetdevWithPortNameCriteria(criteria func(string) bool) (string, error)
 	return "", fmt.Errorf("no representor matched criteria")
 }
 
+// Finds the VF index of a port, if applicable
+func GetVfIndexFromNetDevice(netdev string) (int, error) {
+	if !isSwitchdev(netdev) {
+		return 0, fmt.Errorf("netdev %s is not an eswitch port", netdev)
+	}
+
```

Comment:
add check to ensure the given representor is VF ?

you can use GetRepresentorPortFlavour()